salt/_modules/metalk8s_package_manager.py

```diff
@@ -0,0 +1,56 @@
+'''
+Describes our custom way to deal with yum packages
+so that we can support downgrade in metalk8s
+'''
+
+import re
+import logging
+
+from salt.exceptions import CommandExecutionError
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'metalk8s_package_manager'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def list_pkg_deps(pkg_name, version):
+    '''
+    Check dependencies related to the packages installed so that we can pass
+    this information to pkg.installed
+
+    name
+        Name of the package installed
+
+    version
+        Version number of the package
+
+        Use : salt '*' metalk8s_package_manager.list_pkg_deps kubelet 1.11.9
+    '''
+    log.info(
+        'Listing deps for "%s" with version "%s"',
+        str(pkg_name),
+        str(version)
+    )
+    if not version:
+        return {pkg_name: None}
+
+    pkgs_list = {pkg_name: version}
+    deps_list = __salt__['cmd.run_all'](
+        [
+            'repoquery', '-q', '--qf', '%{NAME} %{VERSION}', '--installed',
+            '--whatrequires', '{}-{}'.format(str(pkg_name), str(version))
```

Comment:
Hmm for me. That doesn't works as you need to enable the repo `--enablerepo=...` and also the package is holded here so you will always get empty answer.
Also `--installed` doesn't work on my side. Always got empty answer (btw it's works since we use kubelet-1.11.10 that require the same version of kubernetes-cni as the kubelet 1.12.6